sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionProcessorFactoryImpl.java

```diff
@@ -60,20 +93,24 @@ public PartitionProcessor create(Lease lease, ChangeFeedObserver observer) {
             throw new IllegalArgumentException("lease");
         }
 
-        String startContinuation = lease.getContinuationToken();
-
-        if (startContinuation == null || startContinuation.isEmpty()) {
-            startContinuation = this.changeFeedProcessorOptions.getStartContinuation();
+        FeedRangeInternal feedRange = new FeedRangePartitionKeyRangeImpl(lease.getLeaseToken());
+        ChangeFeedState state;
+        if (Strings.isNullOrWhiteSpace(lease.getContinuationToken())) {
+            state = new ChangeFeedStateV1(
```

Comment:
ChangeFeedStateV1 will always expect the ChangeFeedMode to be "INCREMENTAL". Rather than creating the FeedRangePartitionKeyRange here, we can make it as part of the ChangeFeedStateV1 constructor and instead pass the getStartFromSettings() as the parameter for the constructor of ChangeFeedStateV1.

Reply:
No - because ChangeFeedStateV1 is used across CFP and Pull model - it is the representation of teh Changefeed query pull continuation. In CFP we map it to the existing Lease format - which has some additional constraints (like always being INCREMENTAL, currently always having a PKRangeId FeedRaneg etc.) - but the ChangeFeedStateV1 is a superset - and even for the CFP we will need to change the Lease contract to allow more flexible feed ranges when enabling merge support.